tests/storage/migration.go

```diff
@@ -587,6 +591,132 @@ var _ = SIGDescribe("Volumes update with migration", decorators.RequiresTwoSched
+				dv := createDV()
+				libstorage.CreateFSPVC(destPVC, ns, sizeWithOverhead, nil)
+				vm := createAndStartVM(dv)
+
+				By("Update volumes")
+				updateVMWithPVC(vm, volName, destPVC)
+
+				var migration *v1.VirtualMachineInstanceMigration
+				Eventually(func() v1.VirtualMachineInstanceMigrationPhase {
+					ls := labels.Set{
+						virtv1.VolumesUpdateMigration: vm.Name,
+					}
+					migList, err := virtClient.VirtualMachineInstanceMigration(ns).List(context.Background(),
+						metav1.ListOptions{
+							LabelSelector: ls.String(),
+						})
+					Expect(err).ToNot(HaveOccurred())
+					if len(migList.Items) < 1 {
+						return v1.MigrationPhaseUnset
+					}
+					Expect(migList.Items).To(HaveLen(1))
+					migration = &migList.Items[0]
+					return migration.Status.Phase
+				}).WithTimeout(time.Minute).WithPolling(time.Second).Should(Equal(v1.MigrationRunning))
+
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name,
+					metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Killing the source pod")
+				pod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+				err = virtClient.CoreV1().Pods(vmi.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{
+					GracePeriodSeconds: pointer.P(int64(0)),
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Waiting for the migration to be set as failed")
+				Eventually(func() v1.VirtualMachineInstanceMigrationPhase {
+					ls := labels.Set{
+						virtv1.VolumesUpdateMigration: vm.Name,
+					}
+					migList, err := virtClient.VirtualMachineInstanceMigration(ns).List(context.Background(),
+						metav1.ListOptions{
+							LabelSelector: ls.String(),
+						})
+					Expect(err).ToNot(HaveOccurred())
+					if len(migList.Items) < 1 {
+						return v1.MigrationPhaseUnset
+					}
+					Expect(migList.Items).To(HaveLen(1))
+					migration = &migList.Items[0]
+					return migration.Status.Phase
+				}).WithTimeout(time.Minute).WithPolling(time.Second).Should(Equal(virtv1.MigrationFailed))
+
+				virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Start(context.Background(), vm.Name, &virtv1.StartOptions{Paused: false})
+				checkVolumeMigrationOnVM(vm, volName, dv.Name, destPVC)
+
+				By("By removing the update volume strategy in order to remove the manual recovery condition and start the VM")
+				p, err := patch.New(
+					patch.WithRemove("/spec/updateVolumesStrategy"),
+				).GeneratePayload()
+				Expect(err).ToNot(HaveOccurred())
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, p, metav1.PatchOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() bool {
+					vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
```

Comment:
Should we verify that `VirtualMachineManualRecoveryRequired` exists before it's removed?

Reply:
The name of the function is unfortunate, not sure what I thought when I did that, but it is done by `checkVolumeMigrationOnVM` (see [here](https://github.com/kubevirt/kubevirt/blob/main/tests/storage/migration.go#L242))